apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.RequestTemplate.cs

```diff
@@ -115,18 +115,34 @@ public sealed class RequestTemplate
             /// </remarks>
             public IReadOnlyDictionary<string, IReadOnlyCollection<string>> ContentHeaders { get; }
 
+            /// <summary>
+            /// The query parameters which will be included with the request. This will never be null.
+            /// If null is specified in <see cref="WithQueryParameters(IEnumerable{KeyValuePair{string, IEnumerable{string}}})"/>
+            /// then an empty dictionary will be used.
+            /// </summary>
+            /// <remarks>
+            /// When using <see cref="SigningVersion.V2"/> query parameters are not included in the signature
+            /// and <see cref="UrlSigner.Sign(RequestTemplate, Options)"/> and
+            /// <see cref="UrlSigner.SignAsync(RequestTemplate, Options, System.Threading.CancellationToken)"/>
+            /// will throw <see cref="ArgumentException"/> if <see cref="RequestTemplate.QueryParameters"/> is not
+            /// an empty collection.
```

Comment:
This is a little confusing to me because `The query parameters which will be included with the request. This will never be null.` where as this states: `When using <see cref="SigningVersion.V2"/> query parameters are not included in the signature... will throw <see cref="ArgumentException"/> if <see cref="RequestTemplate.QueryParameters"/> is not an empty collection`

Reply:
But a null reference is not the same as a reference to an empty collection.

- `RequestTemplate.QueryParameters` will never be null because even if a null is passed to `WithQueryParameters`, an empty dictionary will be set. This is basically to make calling code cleaner by not having to deal with the special case "null".
- When using V2, if custom query parameters are specified, that is, if `RequestTemplate.QueryParameters` is referencing other than an empty collection (it will never be a null reference because of the previous point) then the signing method will throw.

I've added a short phrase at the end of the comment to try and help clarify, or let me know exactly what is confusing for you and what wording would help make it clearer.